server/public/pluginapi/kv_memory.go

```diff
@@ -0,0 +1,284 @@
+package pluginapi
+
+import (
+	"bytes"
+	"encoding/json"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+	"unicode/utf8"
+
+	"github.com/mattermost/mattermost/server/public/model"
+
+	"github.com/pkg/errors"
+)
+
+// MemoryStore is a implementation of the plugin KV store API for testing.
+// It's not meant for production use.
+// It is safe for concurrent use by multiple goroutine.
+type MemoryStore struct {
+	mux   sync.RWMutex
+	elems map[string]elem
+}
+
+type elem struct {
+	value     []byte
+	expiresAt *time.Time
+}
+
+func (e elem) isExpired() bool {
+	return e.expiresAt != nil && e.expiresAt.Before(time.Now())
+}
+
+// Set stores a key-value pair, unique per plugin.
+// Keys prefixed with `mmi_` are reserved for internal use and will fail to be set.
+//
+// Returns (false, err) if DB error occurred
+// Returns (false, nil) if the value was not set
+// Returns (true, nil) if the value was set
+func (s *MemoryStore) Set(key string, value any, options ...KVSetOption) (bool, error) {
```

Comment:
If changes are made to the production version, what prompts us to keep this implementation in sync?